gapic-common/lib/gapic/rest/error.rb

```diff
@@ -19,15 +19,28 @@ module Gapic
   module Rest
     # Gapic REST exception class
     class Error < ::Gapic::Common::Error
-      # @return [Integer] the http status code for the error
+      # @return [Integer, nil] the http status code for the error
       attr_reader :status_code
+      # @return [Object, nil] the text representation of status as parsed from the response body
+      attr_reader :status
+      # @return [Object, nil] the details as parsed from the response body
+      attr_reader :details
+      # @return [Object, nil] the headers of the REST error
+      attr_reader :headers
+      # The Cloud error wrapper expect to see a `header` property
+      alias header headers
 
       ##
       # @param message [String, nil] error message
       # @param status_code [Integer, nil] HTTP status code of this error
+      # @param details [Object, nil] Details data of this error
+      # @param headers [Object, nil] Http headers data of this error
       #
-      def initialize message, status_code
+      def initialize message, status_code, status, details, headers
```

Comment:
This is a breaking change because the constructor is public. I recommend changing these to keyword arguments and providing a default of `nil`.

Reply:
:+1: